riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/DefaultRiptideRegistrar.java

```diff
@@ -395,19 +394,20 @@ private Tag clientName(final String id, final Client client) {
     }
 
     private BeanMetadataElement trace(final String executor) {
-        final AtomicReference<BeanMetadataElement> result = new AtomicReference<>();
-
-        ifPresent("org.zalando.tracer.concurrent.TracingExecutors", () -> {
-            if (registry.isRegistered("tracer")) {
-                result.set(genericBeanDefinition(TracingExecutors.class)
-                        .setFactoryMethod("preserve")
-                        .addConstructorArgReference(executor)
-                        .addConstructorArgReference("tracer")
-                        .getBeanDefinition());
-            }
-        });
-
-        return Optional.ofNullable(result.get()).orElseGet(() -> ref(executor));
+        final Optional<BeanMetadataElement> result = ifPresent("org.zalando.tracer.concurrent.TracingExecutors",
+                () -> {
+                    if (registry.isRegistered("tracer")) {
+                        return genericBeanDefinition(TracingExecutors.class)
+                                .setFactoryMethod("preserve")
+                                .addConstructorArgReference(executor)
+                                .addConstructorArgReference("tracer")
+                                .getBeanDefinition();
+                    } else {
+                        return null;
```

Comment:
I'd go with `ref(executor)` here. It's a different case. We have two fallbacks now: a) tracer not on the classpath and b) tracer not registered/configured.

Reply:
yes, I wanted to keep it DRY. The null returned here will lead to an empty Optional and trigger the fallback in line 410.